src/parse/parser_public.ts

```diff
@@ -197,13 +197,21 @@ export interface IParserConfig {
      *   - Providing special error messages under certain conditions - missing semicolons
      */
     errorMessageProvider?: IErrorMessageProvider
+
+    /**
+     * During cst analysis, the standard buildInitDefFunc uses "new Function" 
+     * which will trigger unsafe-eval warnings in some web contexts.
+     * You can use this optional flag to avoid the warning.
+     */
+    avoidDynamicCodeDuringAnalysis?: boolean
```

Comment:
We should probably give this flag a more general name:
"avoidDynamicCode" / "avoidEval" / "noEvalCompatible"

In case other optimizations would use these patterns and need to be disabled.